### PR TITLE
 Follow up on shared view

### PR DESCRIPTION
**Pull request by mustafapc19**
Ref: #3038 

What's changed:
 - Fixed redirections issues where shared view will be redirected to login page if not opened in incognito.
 - Added project title to header.
 - Removed default layout from the shared view which was causing double header issue.
 - Improved showing ask password modal logic.
 - Removed show system field check box from shared view.
 - Disabled context-menu.
 - Fixed reordering issue with columns.
 - Added readonly mode to `Checkbox` cell.
 - Added password toggle to shared view modal.

### CODE REVIEW

**Review by dstala — APPROVED** [review]

File: packages/nc-gui-v2/components/cell/Checkbox.vue [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <script setup lang="ts">
 import { ColumnInj, IsFormInj, getMdiIcon, inject } from '#imports'
-import { EditModeInj } from '~/context'
+import { ReadonlyInj } from '~/context'
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/cell/Checkbox.vue:3]: -> `'#imports'`

**Review by pranavxc — APPROVED** [review]

File: packages/nc-gui-v2/composables/useViewColumns.ts [hunk 2]
```diff
@@ -128,16 +121,16 @@ export function useViewColumns(view: Ref<ViewType> | undefined, meta: ComputedRe
...
+        (acc: ColumnType, curr: ColumnType) => ({
           ...acc,
           [curr.id!]: curr,
-        }
-      }, {}) || {}
-    )
-  })
+        }),
+        {},
+      ) || {},
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/composables/useViewColumns.ts:132]: Would probably be more readable to just do a quick check before.
Then you could also remove the `?` modifiers.

```ts
if (!meta.value?.columns) {
return {}
}
```

File: packages/nc-gui-v2/pages/[projectType]/view/[viewId].vue [hunk 3]
```diff
@@ -9,11 +10,13 @@ import {
   useRoute,
   useSharedView,
 } from '#imports'
+import { extractSdkResponseErrorMsg } from '~/utils'
 
 definePageMeta({
   public: true,
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/pages/[projectType]/view/[viewId].vue:16]: Can you remove this? I accidently added this parameter but you seem to use another one 😄

File: packages/nc-gui-v2/pages/[projectType]/view/[viewId].vue [hunk 4]
```diff
@@ -9,11 +10,13 @@ import {
   useRoute,
   useSharedView,
 } from '#imports'
+import { extractSdkResponseErrorMsg } from '~/utils'
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/pages/[projectType]/view/[viewId].vue:13]: -> `'#imports'`

**Review by bcakmakoglu — CHANGES_REQUESTED** [review]

**Pull request merged**